tests/unit/modules/test_firewalld.py

```diff
@@ -40,8 +35,73 @@ def test_list_zones(self):
         """
         Test for List everything added for or enabled in all zones
         """
-        with patch.object(firewalld, "__firewall_cmd", return_value=[]):
```

Comment:
I still think there is value in checking what happens if firewall command does not return anything. Could we include this and the test below you edited to be in a seperate test where we are testing an empty response.

Reply:
I added a test for empty response in `list_zones` and restored the one for `list_all`. 

But this specific test was originally wrong, probably by some copy'n'paste error, it said that will test `list_zones` but calls `default_zone`. With this mistake, the original tests didn't tests `list_zones`. The code path for an empty `default_zone` is already tested in the `test_default_zone`, which only returns the `__firewall_cmd` output as-is. I don't think it's needed.